### PR TITLE
Make a basic User model (including secure passwords)

### PR DESCRIPTION
**Pull request by hongsondhs**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+	before_save { self.email = email.downcase }
+	validates :name, presence: true, length: {maximum: 50} 
+	VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+	validates :email, presence: true, length: {maximum: 255}, 
+					  format: { with: VALID_EMAIL_REGEX },
```
Comment (thangtx88) [inline app/models/user.rb]: ident

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+					  uniqueness: {case_sensitive: false}
```
Comment (thangtx88) [inline app/models/user.rb]: ident

File: app/views/static_pages/contact.html.erb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<% provide :title, "Contact" %>
+<% provide :title, (t :contact) %>
 <h1>Contact</h1>
```
Comment (thangtx88) [inline app/views/static_pages/contact.html.erb]: use I18n if you can

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+	before_save { self.email = email.downcase }
+	validates :name, presence: true, length: {maximum: 50} 
+	VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+	validates :email, presence: true, length: {maximum: 255}, 
+	format: { with: VALID_EMAIL_REGEX },
+		uniqueness: {case_sensitive: false}
```
Comment (thangtx88) [inline app/models/user.rb]: ident

Comment (thangtx88) [inline app/models/user.rb]: ident

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+	has_secure_password
+		validates :password, presence: true, length: {minimum: 6}
```
Comment (thangtx88) [inline app/models/user.rb]: ident

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -1,3 +1,3 @@
-<% provide :title, "Sign up" %>
+<% provide :title, t(:signUp) %>
 <h1>Sign up</h1>
```
Comment (thangtx88) [inline app/views/users/new.html.erb]: I18n

File: app/views/users/new.html.erb [hunk 7]
```diff
@@ -1,3 +1,3 @@
...
 <p>This will be a signup page for new users.</p>
```
Comment (thangtx88) [inline app/views/users/new.html.erb]: I18n

File: app/views/static_pages/contact.html.erb [hunk 8]
```diff
@@ -1,5 +1,5 @@
-<% provide :title, "Contact" %>
-<h1>Contact</h1>
+<% provide :title, (t :contact) %>
+<h1><%=t :contact%></h1>
 <p>
   Contact the Ruby on Rails Tutorial about the sample app at the
   <a href="http://www.railstutorial.org/contact">contact page</a>.
```
Comment (thangtx88) [inline app/views/static_pages/contact.html.erb]: I18n

File: app/views/static_pages/contact.html.erb [hunk 9]
```diff
@@ -1,5 +1,5 @@
-<% provide :title, "Contact" %>
-<h1>Contact</h1>
+<% provide :title, (t :contact) %>
+<h1><%=t :contact%></h1>
 <p>
   Contact the Ruby on Rails Tutorial about the sample app at the
```
Comment (thangtx88) [inline app/views/static_pages/contact.html.erb]: I18n

File: app/models/user.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
+  validates :name, presence: true, length: {maximum: 50} 
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255}, 
+  format: { with: VALID_EMAIL_REGEX },
```
Comment (thangtx88) [inline app/models/user.rb]: format: {with: VALID_EMAIL_REGEX},

File: app/models/user.rb [hunk 11]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
```
Comment (thangtx88) [inline app/models/user.rb]: before_save {self.email = email.downcase}

File: app/controllers/application_controller.rb [hunk 12]
```diff
@@ -31,6 +31,6 @@ def default_url_options
 	end
 
 	def hello
-		render html: "he cmn lo!"
+		render html: (t :hello)
```
Comment (thangtx88) [inline app/controllers/application_controller.rb]: Ident

File: app/views/users/new.html.erb [hunk 13]
```diff
@@ -1,3 +1,3 @@
-<% provide :title, "Sign up" %>
-<h1>Sign up</h1>
+<% provide :title, t(:signUp) %>
+<h1><%=t(:signUp)%></h1>
 <p>This will be a signup page for new users.</p>
```
Comment (nhunght) [inline app/views/users/new.html.erb]: i18n

File: app/views/users/new.html.erb [hunk 14]
```diff
@@ -1,3 +1,3 @@
-<% provide :title, "Sign up" %>
-<h1>Sign up</h1>
+<% provide :title, t(:signUp) %>
+<h1><%=t(:signUp)%></h1>
```
Comment (nhunght) [inline app/views/users/new.html.erb]: convention

File: db/migrate/20160922032237_add_index_to_users_email.rb [hunk 15]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToUsersEmail < ActiveRecord::Migration[5.0]
+  def change
+  	add_index :users, :email, unique: true
```
Comment (nhunght) [inline db/migrate/20160922032237_add_index_to_users_email.rb]: indent

**Pull request merged**